### PR TITLE
Update AMS module. Add GA SDK commands.

### PR DESCRIPTION
**Pull request by NahuelBerg**
GA release of Azure Media Services CLI module. 

### Account
- `az ams account check-name`
### Account Filter
- `az ams account-filter create`
- `az ams account-filter delete`
- `az ams account-filter list`
- `az ams account-filter show`
- `az ams account-filter update`
### Asset Filter
- `az ams asset-filter create`
- `az ams asset-filter delete`
- `az ams asset-filter list`
- `az ams asset-filter show`
- `az ams asset-filter update`
### Job
- `az ams job update`
### Asset
- `az ams asset get-encryption-key`
- `az ams asset get-streaming-locators`
### Content Key Policy
- `az ams content-key-policy create`
- `az ams content-key-policy delete`
- `az ams content-key-policy list`
- `az ams content-key-policy show`
- `az ams content-key-policy update`
- `az ams content-key-policy option add`
- `az ams content-key-policy option remove`
- `az ams content-key-policy option update`
### Live Event
- `az ams live-event create`
- `az ams live-event delete`
- `az ams live-event list`
- `az ams live-event show`
- `az ams live-event update`
- `az ams live-event start`
- `az ams live-event stop`
- `az ams live-event reset`
### Live Output
- `az ams live-output create`
- `az ams live-output delete`
- `az ams live-output list`
- `az ams live-output show`
### Streaming Endpoint
- `az ams streaming-endpoint create`
- `az ams streaming-endpoint delete`
- `az ams streaming-endpoint list`
- `az ams streaming-endpoint scale`
- `az ams streaming-endpoint show`
- `az ams streaming-endpoint start`
- `az ams streaming-endpoint stop`
- `az ams streaming-endpoint update`
- `az ams streaming-endpoint akamai add`
- `az ams streaming-endpoint akamai remove`
### Streaming Locator
- `az ams streaming-locator get-content-keys`
### Media Reserved Units
- `az ams account mru set`
- `az ams account mru show`

### Streaming Locator commands
- `az ams streaming locator` base command replaced with `az ams streaming-locator`
- `--streaming-locator-id` and ` --alternative-media-id support` arguments added
- `--content-keys argument` argument updated
- `--content-policy-name` renamed to `--content-key-policy-name`
### Streaming Policy commands
- `az ams streaming policy` base command replaced with `az ams streaming-policy`
- Encryption parameters support in `az ams streaming-policy create` added
### Transform commands
- `--preset-names` argument replaced with `--preset`. Now you can only set 1 output/preset at a time (to add more you have to run `az ams transform output add`). Also, you can set custom StandardEncoderPreset by passing the path to your custom JSON
- `az ams transform output remove` can be performed by passing the output index to remove
- `--relative-priority`, `--on-error`, `--audio-language` and `--insights-to-extract` arguments added in `az ams transform create` and `az ams transform output add` commands
### Job commands
- `--correlation-data` and `--label` arguments added
- `--output-asset-names ` renamed to `--output-assets`. Now it accepts a space-separated list of assets in 'assetName=label' format. An asset without label can be sent like this: 'assetName='
### Asset commands
- `--storage-account` and `--container` arguments added
- Default values for expiry time (Now+23h) and permissions (Read) in `az ams asset get-sas-url` command added

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by NahuelBerg** [comment]
Please, consider setting `New Command(s)`, `AMS` and `Media Services` labels.
Also, add @johndeu as reviewer.
We are targeting https://github.com/Azure/azure-cli/milestone/49 FYI.

Let us know if you have any questions.

Thanks!

**Comment by damatojuan** [comment]
working on fixing CI errors

**Comment by tjprescott** [comment]
Moving a module from preview to GA required a dedicated command review.

**Comment by tjprescott** [comment]
@Renkon try this (modified from `test_dns_commands.py` in the Network module):
```Python
TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
os.path.join(TEST_DIR, 'data', 'clientAccessPolicy.xml')
```

**Comment by NahuelBerg** [comment]
@tjprescott Travis CI verification passed!

**Comment by damatojuan** [comment]
Following up on this... @tjprescott are you (or someone else) able to take a look at this? Thanks!

File: src/command_modules/azure-cli-ams/setup.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 DEPENDENCIES = [
     'azure-cli-core',
-    'azure-mgmt-media==1.0.0rc1',
+    'azure-mgmt-media==1.0.0',
```
Comment (lmazuel) [inline src/command_modules/azure-cli-ams/setup.py]: Please update to 1.0.1, there is a package trouble in 1.0.0 that prevents installation of the CLI in Homebrew (@tjprescott )

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/account_filter.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            tracks_json = json.loads(tracks)
+            for track_selection_json in tracks_json:
+                track_properties = []
+                for track_property_json in track_selection_json.get('trackSelections'):
+                    track_property = FilterTrackPropertyCondition(**track_property_json)
+                    track_properties.append(track_property)
+                tracks_result.append(FilterTrackSelection(track_selections=track_properties))
+        except TypeError as ex:
+            errorMessage = 'Malformed JSON.'
+            raise CLIError('{}. {}'.format(str(ex), errorMessage))
```
Comment (42B) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/account_filter.py]: I'm curious what the definition of "Malformed JSON" is in this case but it seems like maybe `FilterTrackPropertyCondition` and/or `FilterTrackSelection` are the ones that would raise a `TypeError`? The reason I ask is that `json.loads` usually throws either a `ValueError` in Python 2 or `JSONDecodeError` (subclass of `ValueError`) in Python 3.

* https://docs.python.org/2/library/json.html#json.loads
* https://docs.python.org/3/library/json.html#json.loads
```python
>>> import json
>>> json.loads('{')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)

```
Of course a `TypeError` would be raised if `tracks` was None but the `if` statement above precludes that possibility?

```python
>>> json.loads(None)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not NoneType
```

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_utils.py [hunk 3]
```diff
@@ -17,3 +22,63 @@ def _is_guid(guid):
...
+    return iso_duration_format_value
+
+
+def parse_timedelta(str_duration):
+    if str_duration:
+        datetime_duration = datetime.strptime(str_duration, '%H:%M:%S')
+    return timedelta(hours=datetime_duration.hour or 0,
+                     minutes=datetime_duration.minute or 0,
+                     seconds=datetime_duration.second or 0)
+
```
Comment (42B) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_utils.py]:
```python
>>> from datetime import datetime, timedelta
... 
... 
... def parse_timedelta(str_duration):
...     if str_duration:
...         datetime_duration = datetime.strptime(str_duration, '%H:%M:%S')
...     return timedelta(hours=datetime_duration.hour or 0,
...                      minutes=datetime_duration.minute or 0,
...                      seconds=datetime_duration.second or 0)
... 
>>> parse_timedelta('')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 7, in parse_timedelta
UnboundLocalError: local variable 'datetime_duration' referenced before assignment
```
Also, what is the purpose of the three occurrences of `or 0`?

```python
>>> def parse_timedelta2(str_duration):
...     if str_duration:
...         datetime_duration = datetime.strptime(str_duration, '%H:%M:%S')
...     return timedelta(hours=datetime_duration.hour,
...                      minutes=datetime_duration.minute,
...                      seconds=datetime_duration.second)
... 
>>> parse_timedelta('0:0:0')
datetime.timedelta(0)
>>> parse_timedelta2('0:0:0')
datetime.timedelta(0)
```

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/asset_filter.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            tracks_json = json.loads(tracks)
+            for track_selection_json in tracks_json:
+                track_properties = []
+                for track_property_json in track_selection_json.get('trackSelections'):
+                    track_property = FilterTrackPropertyCondition(**track_property_json)
+                    track_properties.append(track_property)
+                tracks_result.append(FilterTrackSelection(track_selections=track_properties))
+        except TypeError as ex:
+            errorMessage = 'Malformed JSON.'
+            raise CLIError('{}. {}'.format(str(ex), errorMessage))
```
Comment (42B) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/asset_filter.py]: Same as above, `JSONDecodeError` and/or `ValueError`?

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_locator.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    def __valid_content_key(key):
+        return key.get('id') and key.get('value') and key.get('labelReferenceInStreamingPolicy')
+
+    obj = None
+
+    try:
+        obj = json.loads(content_keys)
+    except ValueError as err:
+        raise CLIError('Malformed JSON: ' + str(err))
```
Comment (42B) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_locator.py]: This one catches a `ValueError` and still describes it as "Malformed JSON"

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_policy.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        clear_tracks_result = []
+        try:
+            clear_tracks_json = json.loads(clear_tracks)
+            for track_selection_json in clear_tracks_json:
+                track_properties = []
+                for track_property_json in track_selection_json.get('trackSelections'):
+                    track_property = TrackPropertyCondition(**track_property_json)
+                    track_properties.append(track_property)
+                clear_tracks_result.append(TrackSelection(track_selections=track_properties))
+        except TypeError as ex:
```
Comment (42B) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_policy.py]: Similar?

File: src/command_modules/azure-cli-ams/HISTORY.rst [hunk 7]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+2.0.0
++++++
+* GA release.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-ams/HISTORY.rst]: There are too many new commands added to call this module GA (it is irrelevant if the service is GA). Please update to 0.3.0 and specifically call out the new command groups and the breaking changes to existing command groups.

File: src/command_modules/azure-cli-ams/setup.py (hunk 1 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-ams/setup.py]: Agreed. You must update this to 1.0.1.

**Comment by damatojuan** [comment]
working on the changes requested here

**Comment by tjprescott** [comment]
Opened issue https://github.com/Azure/azure-cli/issues/7718 to track remaining GA items.

**Pull request merged**